### DISCUSSION TITLE
Can Floris be used to calculate AEP in 2D?

### DISCUSSION THREAD

**Discussion by victormendoza1983** [NatLabRockies/floris#273] [Q&A] [answered]
Hello all!
I am newbie with FLORIS, and when I run AEP calculations for time series (20 years data with 1 hour average), it takes quite significant time compared to other solvers (few seconds). 
As long as I understand, on of the main reasons can be that FLORIS considers the wake (flow) in 3D... is there any way to makes calculation in 2D in order to reduce computational time?
Best regards,

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi Victor,

Indeed, FLORIS by default calculates the wind speed at every rotor for 25 points. We do this to ensure accuracy, since the wind speed may vary significantly across the rotor. However, for your purposes, you can reduce this to a single point to speed up your computations. You can do this by changing the ``ngrid`` parameter in your .json file:
```
"ngrid": 1,
```
which effectively yields a ``1 x 1`` grid, thus one rotor point.

There are a number of additional steps you can take to speed-up your code. Firstly, you can consider the suggestion made in [this discussion post](https://github.com/NREL/floris/discussions/270) if you are not interested in wake deflection. Secondly, you can also parallelize your code using ``mpi4py`` or ``multiprocessing``.

Also, note that @bayc is currently working on the next iteration of FLORIS, where we specifically target speed improvements for AEP calculations. The aimed release date is some time later this month.

**Reply by victormendoza1983** [reply]
Hello @Bartdoekemeijer 
I tried to implement your suggestion about "ngrid": 1, with no success. As I am newbie, probably I am doing it wrong. Could you explain it to me a bit more in detail or even better... write me some example lines about how to implement it on the json file? It would be really useful for me. Thanks in advance!

![Screenshot 2022-01-04 133520](https://user-images.githubusercontent.com/97036019/148067502-1986d157-44e3-4633-be7d-282ccecc7837.png)

**Comment by victormendoza1983** [comment]
Thanks so much for fast and kind reply!
